pilot/pkg/model/push_context.go

```diff
@@ -1357,10 +1367,19 @@ func (ps *PushContext) updateContext(
 	}
 
 	// Must be initialized in the end
-	// Sidecars need to be updated if services, virtual services, destination rules, or the sidecar configs change
+	// Sidecars need to be updated if services, virtual services, destination rules, or the sidecar configs change.
 	if servicesChanged || virtualServicesChanged || destinationRulesChanged || sidecarsChanged {
-		if err := ps.initSidecarScopes(env); err != nil {
-			return err
+		if !gatewayAPIChanged {
```

Comment:
```suggestion
		if gatewayAPIChanged {
```
Shouldn't it be inverted?